test/lib/util.test.js

```diff
@@ -155,14 +155,22 @@ describe('download_img', function() {
     });
 });
 
-// FIXME this text intermittently fails because it reaches out to an external
-// server. We should replace it with gzipped content served locally
+// FIXME these tests intermittently fails because they reach out to an external
+// server. We should replace them with content served locally
 describe('fetch_html', () => {
+    const doctype = '<!DOCTYPE html>';
+
+    it('works', function () {;
```

Comment:
This new "works" test works for me. I run the tests with: `./node_modules/.bin/mocha --recursive --check-leaks`.

```
  fetch_html
    ✓ works (1757ms)
```
So I suggest moving the comment about "This test fails" to the "can handle gzipped responses" instead of applying it to the two "fetch html" tests. Or further, either comment or remove the failing test.

Reply:
Ah no I was wrong, it is in fact intermittently working/failing! But after looking at the error message `For async tests and hooks, ensure "done()" is called` and looking at mocha docs, I found a way to make these tests work:

```
    it('works', done => {
        const test_url = 'https://creativecommons.org/blog';
        return util.fetch_html(test_url).then(result => {
            expect(result.html().substring(0, doctype.length)).to.equal(doctype);
        }).finally(done());
    });

    it('can handle gzipped responses', done => {
        const test_url = 'https://www.kapanlagi.com/' +
                        'intermezzone/' +
                        'bule-amerika-ini-nyoba-makan-buah-duku-ekspresinya-nggak-nahan-aee243.html';
        return util.fetch_html(test_url).then((result) => {
            expect(result.html().substring(0, doctype.length)).to.equal(doctype);
        }).finally(done());
    });
```

Reply:
Also we are favouring arrow functions `=>` instead of `function ()`.

Reply:
Are arrow functions a problem in Mocha tests? In Jasmine tests, which look very similar, I avoid them because Jasmine can use the `this` object passed to the test functions to store state (although that is optional.)